.github/workflows/app-publish.yaml

```
@@ -0,0 +1,121 @@
+name: App Publish
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release*
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    if: ${{ startsWith(github.ref, 'refs/tags/v') || github.repository == 'jellyfin/jellyfin-android' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set JELLYFIN_VERSION
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
+
+      - name: Build Release
+        run: ./gradlew --no-daemon --build-cache assembleRelease versionTxt
+
+      - name: Split APK release types
+        run: |
+          mkdir app/build/outputs/apk/libre_release app/build/outputs/apk/proprietary_release;
```

```suggestion
          mkdir build/jellyfin-publish/release-libre build/jellyfin-publish/release-proprietary;
```

I'd rather use a (custom) build directory in the root

---

The version.txt should be moved to `build/jellyfin-publish/`, and that directory will be the artifact path